modules/security/src/main/java/io/oasp/module/security/common/api/accesscontrol/AbstractAccessControl.java

```diff
@@ -84,7 +84,7 @@ public boolean equals(Object obj) {
     if (getClass() != obj.getClass()) {
       return false;
     }
-    AccessControl other = (AccessControl) obj;
+    AbstractAccessControl other = (AbstractAccessControl) obj;
```

Comment:
For me the better options is to have:
- An interface to cast to
- An abstract class to provide a helper implementation